llms/mlx_lm/tuner/trainer.py

```diff
@@ -60,25 +60,21 @@ def iterate_batches(dataset, tokenizer, batch_size, max_seq_length, train=False)
         indices = np.random.permutation(indices)
         # Collect batches from dataset
         for i in range(0, len(indices) - batch_size + 1, batch_size):
-            # Encode batch
+            # Encode batch up to the max seq length
             batch = [
-                tokenizer.encode(dataset[indices[i + j]]) for j in range(batch_size)
+                tokenizer.encode(dataset[indices[i + j]])[:max_seq_length]
```

Comment:
@awni I noticed this when I tried to qlora mixtral using the guanaco dataset. The memory spiked and crashed the training, after I changed the iterate_batches to only return a specified max seq length, the training worked as expected.

Reply:
Cool. I would suggest we keep a warning below because some people may have really long paragraphs and the right way to deal with that is not to truncate them here but to split them in a preprocessing step.

So first:
1. Encode full length
2. Warn if length is too long ("longest sentence {max(lengths)} will be truncated to {max_seq_length}"
3. Truncate and make an array batch

Reply:
@awni I updated it based on the suggestion. My current Mac studio is running the training. I didn't do comprehensive testing, but from a quick test on my laptop it looks good.